app/controllers/application_controller.rb

```diff
@@ -2,4 +2,29 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  helper_method :current_user
+
+  before_action :ensure_current_user, only: [:new, :create, :update, :destroy]
+
+  def current_user
+    user = User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def ensure_current_user
+    redirect_to new_session_path unless current_user
+  end
+
+  def log_in(user)
+    session[:user_id] = user.id
```

Comment:
I don't think you need this log_in method - you only use it from sessions_controller and it's a single-liner. You may as well just do that there. Or if you do want an explanatory method name, leave it in that controller rather than the superclass. The superclass is a good place for behavior that all controllers need, but not so for something used in only one.